ml/notebooks/netdata_anomaly_detection_deepdive.ipynb

```diff
@@ -0,0 +1,1705 @@
3ei498Mrm+X3sItWJs6NGiPj5YtAfb07Ix+aEuftsfPcubfXX3F+t07xsADJvi363n5Tnb1W6eWn6Rf/XHTD89NbNAl0l2Tct66HJZPfWxdvHRs7ahYc0oAMD7C/fgmd5X+O3Xw3OREyyW4zlGr728Pp6b4X3+NqsFD11zWeCV3bS/6cPdr9Ha/Rm6fvECyrHvm50Rqqc1d6Ym/LkLMzX1oAZ8ugYp7w7A1tQsvDJnBzYcyAirvsnTU7cg0mrB3rdu1U3ffTwH97oLSnZvXg8zn7pW91453EHFT5buQ0ykVT1RWbD9BBZsP4FBnZupy0op1fme3wSj9+/vXSeRU7RZ/ZwAwBB3N7+lyafwxYoDfsdlINquF573JuXdAeg/abX6t0d8SiY+XbYfO47l4Pth3dTpviPuae+ir9l/Gmv2nzZsk/Z40vp4yV60blQDmT6Bldd9urp6jivPd0jvNg11r6nueQb5HtQGCeYlHUOPlvUweeUBuKTErR0uxvsLvRem/+x6iS54umjnSbRrnIJnb26tTnvAHURJeXcABn+pDCN/ca1odb7nLqnR65KUmoUkg5otr87VP//7ul2CMb9tV2sIPjsjEXdd3TRoFsMHi/b4ZQJ9tnwfRvdri7cXJGNagKL8j3wf79d1T1uj7/5vNuAqcQBQvobgsJcgUrPskMnrUb96JDa7f8sA5WIwJsK8toAngOB5HX1FaEYL3XgoE/9btt/vxpJRfYn/TN9qWoPIU1A5LuI97JdNMMHxkGnXy0BeMwnEHdIElc2+i6asTTHdrrYMgMUiUK3Ye1E3f8U6OOv01y//7caQaos9bRKQ8tDW4bn/G+W775fHewBQbqgdSM/3uyCPW5eiy1C+4zNvd0jPp+hUbpHuXMAGB+61rsBs5/UoQDTqCf8bU7uO52DX8ZyAmbl3fa4ErZrWicH+U3l+mUebD5/B+Dva41v3Bfmgzk3VeYO+MD7ePJlV11m8n0dPZkyUzYI9E/S/E4/94A0YKqPPlWDyygOG2WXFDhfG/LYNK/em43hWEX7acBjZhXaMvbUt/tpxHO/8lYzj2UV4bWB70+ccty4FE+bvhsMlcXnD6vjP9ER13tSNR3TfOaOmb8XqfafRrUU9XN6wBgDj8xht8MLjqZ+3YO6/e+peU89NLM/rfkenJmoQcVr8ERzOKEB0hBV3xXpfZ9/Aqdaek7l4969kLE8+hY2HMlEbeYD7azQCSibQv3/ZYtojQuver9cjNbMQt3VsjAj3eaQ2CPnb5jTsPJatZmNGWAUe79VSXe7/fk1CzWgbYiKsftcsv285qgbNmtbxFuktdriw+3guXv9jF1buTccKzXWeUaYM4O0uHM7v+Id/70XcuhS0bFAd/+zmvRA/kJ6Hd/5KxuJdJ/FC/7aGwXCjGw4bD2Wqwf9YcUb9Pl+223sc/POr9Vg5+qaA7XpvYTKKHU5cd3kDXRbmp8v2Y/HuU/jrP71032URFgtG/MP8vBVQPpcZ+SW4rUPjkAbZeNzd1T/Q63nbp/7nOx5m11Z5xQ71WPxo8V6/siPzko7h7+fMr3+Gac63vPS/yxXZlao8gkBOWH3CQP6jg62T7QEkIxsXUBCo0O70S7ctjYz88IIjKRnmhb20BQV9mRV83ngoE9kFdkhIw+JZh077RzrDGTL3bFuz7zSiI6y4tmV9NdiS5k5fPZxRgF3HcpBdGPxESTv86foAhSwX7zIPAmrvapoVDtdOz8wvwY6j2cjIL8Gmw4F/8FbuTUeEReDqS+sGXK40ktKy0bx+NeQUOdCiQXW4XNI0Y0p7p2DvyVwU210QAqhXPRINa0aFFPzLLrCrhat3HstBXrEDXS6ri8z8EsREWP1GvtlzIhfHs/QXKftP5QbNBDiRXYTsQmVbJ3KKkJpZgGKHC03rxOB4diFaNqyhFuz20HZzMsse2nXc/+TVIyk1Cx2a1sah0/k4477gzSt2YMPBDNStFommdWNM77acyS+B1SoCFmuXUqLE6dIV1zxTUIIT2UVqwCEjrxgbD2agZnQErmxSC4By4tCkdgxiIr0XZgdO6b9b1u47jbYX10RnzTGWmlkAh0v6XdCFkrEthL4QuVZesQPJx3Nwaf1quKim90I6MTULDqcLrS6qqfusmNWX8QS7zV5T5W6xxPHsQlxWvzocThe2ui/GS5wu5BTZUVjihBDARTWjsV2T1bfnZK7fqI4lTpf6vu47mYd0n+9GbaZcwuEz6Ni0ti4L70C68Xvrm2EXrNtIem4xDp3Oh8PpwhUX1YAQAluPnEG7xrUM77ppX5/NhzPRpE4Miu0utT3J7mP6QHoe6lWLDHiMa9ucV+zAFRfVQInD5fda+TqaVejX/fekz/MstDt19XviD2WiT7tGOJJRAIfLhUvqVVPnZeQVY8PBDFx9aR1E2bzHZ2pmgW45D89nOzE1yy+zIaugRD0uPGZtTsPTvS/XbRuAbt0TBsfd6bxipOcWo261SL95wViE8MtmM+tirs1c3HE0G7ViInTz5yYeQ//2jXV3iI0YfbaOZxVi/6k89bk2FFnqvNTl36Lk2mfhdEm1CHxGfomaoelpW51SPH8P7W+wpw1HA5zvbE/LRm6R3TDjVW13ZiGscKK3NQm9kYTPHHchCzVL3caKdCa/BNWd3udyqzUev2feoFvGKAB0KqcIJU5XmYZzB4AUn89pOAH2g+n52H8qz++4HWpdgtcifgQATHXejAjNxVBCivk517oDp5FjcA53JLPAsDbdruM5avd7ILwivZcK//OeYocLmw+fQe0YGxrXjvHrJpSUlq0LWvq3J1v9/vUER45mFSIjrxipmcoxvfNYNralZcFqEWhWtxpyi+xoVrcaih1OrNyTrmanHD1TiOwC4/OS03nFSEg5o97MlFLiRHYR6lSLMFzecBu5xWotQo+1B07jolreAE+m5hrGc1627kAGOjatrU6X7ufYqGaUepPP5ZJYmnwKjWop0QfP+fJlwvv99IRtAe
```

Comment:
```suggestion
    "The dotted lines correspond to the feature vectors sampled from the yellow anomalous period. As such, we expect them to look very different to the solid lines (sampled from the normal period). The two centroids are represented by the solid circled lines. Admittedly it's not the clearest of plots to read since the normal lines all bunch together."
```